telegram/ext/filters.py

```diff
@@ -603,6 +603,66 @@ def filter(self, message: Message) -> bool:
         xml = mime_type('application/xml')
         zip = mime_type('application/zip')
 
+        class file_extension(MessageFilter):
+            """This filter filters documents by their file ending/extension.
+
+            Note:
+                * This Filter only filters by the file ending/extension of the document,
+                    it doesn't check the validity of document.
+                * The user can manipulate the file extension of a document and
+                    send media with wrong types that don't fit to this handler.
+                * Case insensitive by default,
+                    you may change it with the flag ``case_sensitive=True``.
+                * Extension should be passed without leading dot
+                    unless it's a part of the extension.
+                * Pass :obj:`None` to filter files with no extension,
+                    even empty one. E.g. without a dot in the filename.
+
+            Example:
+                ``Filters.document.file_extension("jpg")``
+                    filters files with extension ``".jpg"``.
+                ``Filters.document.file_extension(".jpg")``
+                    filters files with extension ``"..jpg"``.
+                ``Filters.document.file_extension("Dockerfile", case_sensitive=True)``
+                    filters files with extension ``".Dockerfile"`` minding the case.
+            """
+
+            def __init__(self, file_extension: Optional[str], *, case_sensitive: bool = False):
```

Comment:
I don't think the `*` is necessary

Reply:
Calling `Filter.file_extension('jpg', True)` is just insensible.

Reply:
We don't use this kind of signatures anywhere else in the lib and I don't want to add it just in one place. As with `__init__` and lists: either change it everywhere (which is breaking and needs to be discussed), or not at all.
Aside from that, PTB is not about handholding its users.